src/main/java/com/epam/aidial/core/service/ResourceService.java

```diff
@@ -199,30 +150,23 @@ public Pair<ResourceItemMetadata, String> getResourceWithMetadata(ResourceDescri
 
     @Nullable
     public Pair<ResourceItemMetadata, String> getResourceWithMetadata(ResourceDescription descriptor, boolean lock) {
-        String redisKey = redisKey(descriptor);
-        Result result = redisGet(redisKey, true);
+        String redisKey = cacheService.redisKey(descriptor);
+        CacheService.Result<CacheService.Item<String>> result = cacheService.getString(redisKey);
 
         if (result == null) {
             try (var ignore = lock ? lockService.lock(redisKey) : null) {
```

Comment:
Move lockService.lock to CacheService.lock to encapsulate it and use only CacheService in ResourceService and FileService.

Reply:
As discussed, moved the logic of CacheService back to the ResourceService.